extensions/amp-autocomplete/0.1/amp-autocomplete.js

```diff
@@ -115,6 +125,13 @@ export class AmpAutocomplete extends AMP.BaseElement {
         `${TAG} should contain exactly one <input> child`);
     this.inputElement_ = inputElements[0];
 
+    if (this.templates_.hasTemplate(
```

Comment:
there shouldn't be a need to pass the query selector. `hasTemplate(this.element)` should do the right thing ( if it is not, let's fix the root cause)

Reply:
@caroqliu and I discussed this offline. `hasTemplate` doesn't support elements that use both `script[template]` and `script[type="application/json]`. We could fix, but it might be complex. Should we leave a PR and investigate the effort for a followup PR?

Reply:
let's do that as a separate thing since every place that's calling `hasTemplate` (e.g. `amp-list`) is also not behaving correctly.